ユーザ認証機能、各種ユーザページを追加

damisumida opened the pull request:
「devise を使ってユーザー認証を実装する」の課題を作成いたしました。
ご確認をお願いいたします。

damisumida committed:
- deviseのインストール・セットアップ

damisumida committed:
- User modelをセットアップ
- サインアップとログインリンクの追加

damisumida committed:
- Usersテーブルへカラム追加

damisumida committed:
- ``` rails g devise:views user ``` を実行

damisumida committed:
- ``` rails g devise:controllers users ```を実行

damisumida committed:
- debugを追加
- viewへのカスタマイズが反映されるよう、devise.rbを変更
- アカウント編集ページに郵便番号、住所、自己紹介文を記載する欄を追加

damisumida committed:
- サインアップ、サインイン、ログアウト後の遷移先を指定

damisumida committed:
- rails g User controllerを実行
- ユーザ一覧画面を追加
- ユーザ詳細画面を追加

damisumida committed:
- ユーザ情報編集できるようconfigure_permitted_parametersを追加
- ログイン・アウト後の画面遷移をsession_controllerで指定するよう変更
- ユーザ情報編集後、books_pathに遷移するよう変更

damisumida committed:
- letter_opener_webを追加

damisumida committed:
- 日本語訳を追加
- ログインしたIDに一致するユーザのみ編集できるよう変更
- ログインしていない状態でbooks、userページが表示されないよう修正

damisumida committed:
- 不要なファイル・コメントを削除

damisumida committed:
- erblint, rubocopのエラーを修正

damisumida committed:
- 不要なファイルを削除
- パスワード再設定後の画面遷移先を設定
- 画面表示の改行ずれを修正
- 翻訳漏れを修正

damisumida committed:
- ヘッダーメニューにbooksへのリンクを追加

damisumida committed:
- usersテーブルより使用していないカラムnameを削除

damisumida committed:
- 翻訳抜けを修正

yuuu commented on app/controllers/books_controller.rb:
```diff
@@ -67,9 +68,4 @@ def destroy
   end
-
-  # Only allow a list of trusted parameters through.
-  def book_params
```
このメソッドを削除してしまうと、本の作成・更新ができなくなりそうに見えます。

yuuu commented on config/initializers/devise.rb:
```diff
@@ -0,0 +1,311 @@
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
ここはdeviseから送信されるメールの `from` に当たる部分なので `no-reply@example.com` のようなメールアドレスにしておくと良いですね。

yuuu commented on config/initializers/devise.rb:
```diff
@@ -0,0 +1,311 @@
+
+  # ==> Configuration for :validatable
+  # Range for password length.
+  config.password_length = 6..128
```
昨今のセキュリティ情勢を加味すると、ミニマムは8文字の方が良さそうです

yuuu commented on app/controllers/books_controller.rb:
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class BooksController < ApplicationController
+  before_action :authenticate_user!
```
これは全コントローラで呼び出す処理となっているので、親のコントローラにまとめても良さそうですね。

damisumida committed:
送信メールアドレスを変更
パスワードの最小文字を8文字に変更
``` authenticate_user! ```を親コントローラで使用するよう修正
``` book_params ``` メソッドを再追加

damisumida commented:
ご指摘いただいた箇所を修正いたしました。
再度ご確認をよろしくお願いいたします。
